File: 2523-closest-prime-numbers-in-range/2523-closest-prime-numbers-in-range.py
class Solution:
    def closestPrimes(self, left: int, right: int) -> List[int]:

        # we use here algorithim is seive for more optimal way

        if right < 2:
            return [-1,-1]

        prime = [True]*(right+1)
        prime[0]=False
        prime[1]=False

        for i in range(2,int(right**0.5)+1):
            if prime[i]==True:
                for j in range(i*i,right+1,i):
                    prime[j]=False
        num=[]
        count=left
        for i in prime[left:right+1]:
            if i==True:
                num.append(count)
            count+=1
        if len(num)<2:
            return [-1,-1]
        
        final=[0]*2
        res=float('inf')
  
        for i in range(1,len(num)):
            if num[i]-num[i-1]<res:
                
                res=num[i]-num[i-1]
       
                final[0]=num[i-1]
                final[1]=num[i]
        return final
        
            

      
      
      

File: 2523-closest-prime-numbers-in-range/2523-closest-prime-numbers-in-range.java
class Solution {
    public int[] closestPrimes(int left, int right) {
    // same we use seive algorithim here

    int[] res=new int[2];

    res[0]=-1;
    res[1]=-1;

    if (right < 2){
        return res;
        
    }

    boolean[] prime=new boolean[right+1];
    Arrays.fill(prime,true);
    prime[0]=false;
    prime[1]=false;

    for(int i=2;i*i<=right;i++){
        if(prime[i]){
            for(int j=i*i;j<=right;j+=i){
                prime[j]=false;
            }
        }

    }
    ArrayList<Integer> arr=new ArrayList<>();
    for(int i=left;i<=right;i++){
        if(prime[i]){
            arr.add(i);
        }

    }
    if(arr.size()<2){
        return res;
    }
    int count=111111111;
    for(int i=1;i<arr.size();i++){
        int cal=arr.get(i)-arr.get(i-1);
     
        if(cal<count){
            count=cal;
            res[0]=arr.get(i-1);
            res[1]=arr.get(i);
        }
    }
        
    return res;}
}